syllabus/6-Polkadot-SDK/Polkadot-SDK-slides.md

```diff
@@ -0,0 +1,157 @@
+> security of Polkadot.
+
+In this architecture, we call Polkadot the Relay Chain. we call each of these blockchains a
+Parachain.
+
+Each computation unit of the Relay Chain that is capable of progressing a parachain block in
+parallel was called a Core.
+
+Polkadot initially allowed parachains to compete for deployment on Polkadot via auctioned slots.
+Winners get access to a Core for a long long time. Now, it is moving toward a more agile system where
+parachains can purchase cores on-demand.
+
+In some sense, Polkadot is moving towards a general purpose computer, exposing its Core as the
+primary primitive. Whether they are used to build a Parachain or not, it does not care. The future
+of Polkadot is the World Computer vision.
+
+### 2. How Do We Build a Parachain?
+
+Substrate and FRAME
+
+Building a blockchain is damn painful hard. This is why Polkadot has pivoted for years to build a
+modular and extensible blockchain framework called Substrate.
+
+Substrate has been used to build:
+
+1. the mighty Polkadot Relay Chain
+2. all the Polkadot Parachains to date.
+3. (less known) can be used to build any blockchain unrelated to Polkadot (solo-chain).
+
+> There is another framework involved called Cumulus, but since it has a very small footprint, we
+> will skim over it here. Cumulus augments Substrate to make it Polkadot-compatible.
+
+FRAME is a subset of blockchain within Substrate that allow you to focus only on writing the
+state transition function aka. Runtime of your blockchain and not deal with the rest of the
+software. WASM.
+
+### 3. Communication
+
+XCM
+
+Shared Security is GREAT, but it would be even more FAN-TAS-TIQUE is if these parachains could
+communicate with each other. Why is that? **Shared security puts communication on steroids**.
+
+https://x.com/TheDotsMagazine/status/1790755070490857778
+
+Chains can trust the execution of each other to a much higher extent. But note that it is still not
+100% guarantee. The Relay Chain cannot guarantee that a parachain won't upgrade itself to go rogue.
+
+Moreover, the sender and recipient of these messages can be contracts, accounts, or any other
+abstract entities within chains, not just the chain itself. "Consensus System".
+
+Polkadot Relay Chain provides:
+
+1. Two communication protocols for "Consensus Systems" to send payloads to one another (the
+   telephone line).
+   1. XCMP/HRMP
+   2. VPM (UMP, DMP)
+2. One language to compose messages and programs and send them over.
+   1. XCM
+
+Async and Sync: Disjoint consensus systems are in principle[^1] asynchronous by nature. This is an
+important concept to keep in mind.
+
+
+[^1]: Technically, if two parachains are co-scheduled in the same core in the same timeslot, they
+    could have synchronous communication. But they cannot count on this. So at the programming
+    model, one must assume everything is always asynchronous.
```

Comment:
I wouldn't even mention it at that point. Students will not learn about coretime for a long time (near the end of Polkadot) + from what I know co-scheduling two parahcains to same core (in the same slot, so not talking about interlacing) is not in scope for coretime anytime soon if at all. JAM might bring it tho.

Reply:
This will be in the speaker note, I suppose that is fine?